dotnet/Examples/AwsAuth.cs

```
@@ -0,0 +1,91 @@
+        ///     bound_iam_principal_arn="arn:aws:iam::AWS-ACCOUNT-NUMBER:role/AWS-IAM-ROLE-NAME" \
+        ///     ttl=24h
+        ///  Learn more about the available parameters at https://www.vaultproject.io/api/auth/aws#parameters-10
+        /// </summary>
+        public string GetSecretAWSAuthIAM()
+        {
+            var vaultAddr = Environment.GetEnvironmentVariable("VAULT_ADDR");
+            var roleName = Environment.GetEnvironmentVariable("AWS_ROLE_NAME");
+
+            var amazonSecurityTokenServiceConfig = new AmazonSecurityTokenServiceConfig();
+
+            // Initalize BasicAWS Credentials w/ an acessKey and secretKey
+            Amazon.Runtime.AWSCredentials awsCredentials = new BasicAWSCredentials(accessKey: Environment.GetEnvironmentVariable("AWS_ACCESS_KEY_ID"), 
+                                                                secretKey: Environment.GetEnvironmentVariable("AWS_SECRET_ACCESS_KEY"));
+            
+            // Construct the IAM Request and add necessary headers
+            var iamRequest = GetCallerIdentityRequestMarshaller.Instance.Marshall(new GetCallerIdentityRequest());
+            
+            iamRequest.Endpoint = new Uri(amazonSecurityTokenServiceConfig.DetermineServiceURL());
+            iamRequest.ResourcePath = "/";
+
+            iamRequest.Headers.Add("User-Agent", "some-agent");
+            iamRequest.Headers.Add("X-Amz-Security-Token", awsCredentials.GetCredentials().Token);
+            iamRequest.Headers.Add("Content-Type", "application/x-www-form-urlencoded; charset=utf-8");
+
+            try
+            {
+                // Use AWS utility to sign our request
+                new AWS4Signer().Sign(iamRequest, amazonSecurityTokenServiceConfig, new RequestMetrics(), awsCredentials.GetCredentials().AccessKey, awsCredentials.GetCredentials().SecretKey);
+            }
+            catch(Exception e)
+            {
+                Console.WriteLine($"Failed to sign IAM request: ", e.Message);
```

you are swallowing the exception, what is the expected behaviour for the next code block in case of this error? is it valid to use unsigned IAM request?

---

This was a try catch for debugging purposes. Thanks for the catch!